Data Structures/Queue/C++/CircularQueue/CircularQueue.hpp

```diff
@@ -0,0 +1,144 @@
+#ifndef _CIRCULARQUEUE_H_
+#define _CIRCULARQUEUE_H_
+/*
+*@author Ionésio Junior
+*/
+
+#include "resources/Queue.hpp"
+
+/*
+* Queue in circular implementation
+*/
+
+//CircularQueue.hpp
+template<class T>
+class CircularQueue : public Queue<T>{
+	private:
+		T *head_ptr;
+		int tail;
+		int head_index;
+		int size;
+		int elements;
+	public:
+		CircularQueue(int size);
+		bool isEmpty() override;
+		bool isFull() override;
+		void enqueue(T element) override;
+		T* dequeue() override;
+		T* head() override;
+};
+
+//Queue.cpp
```

Comment:
Separate this to a separate CircularQueue.cpp file

Reply:
Done

Reply:
Sorry, what is folder structure format? Data Structures/Circular Queue/C++ and Data Structures/Standard Queue?

Reply:
Yes that works